### PR TITLE
fix(efb): flyPadOS v2 view state fixes

### LINKED ISSUE #4575: EFB UI enhancement
**Is your feature request related to a problem? Please describe.**
I get frustrated when I can not read the efb due to colour scheme.

**Describe the feature you'd like to see implemented**
updated UI with better colour scheme to make easier to read and better icons or tabs.

**References**
efb

**Additional context**

Discord username DarkFury#4936

**BY REMOVING THIS LINE, I ATTEST THAT I HAVE FOLLOWED THE ISSUE TEMPLATE AND PROVIDED A TITLE, as well as searched for other possible duplicate issues. I acknowledge if this line is not removed and/or the issue template is not respected, my issue will be closed without warning**

### PR DESCRIPTION
**Pull request by antonyisoardi**
Fixes #4575

## Summary of Changes

Updates to maintain view state of views when navigating away from them. From now on, when a user is in a "sub-view" for Performance or Dispatch, then will the last selected sub-view when the return to Performance or Dispatch

## Additional context

Discord username (if different from GitHub): Antiso

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

**Comment by Benjozork** [comment]
Note that this needs to be ported over when we switch back to react router

File: src/instruments/src/EFB/Components/Navbar.tsx [hunk 1]
```diff
@@ -1,38 +1,30 @@
-import React, { useState } from 'react';
+import React from 'react';
 
 export type NavbarProps = {
     tabs: string[],
     onSelected: (index: number) => void;
+    activeIndex: number
```
Comment (Benjozork) [inline src/instruments/src/EFB/Components/Navbar.tsx:6]:
```suggestion
    activeIndex: number,
```

**Review by Benjozork — CHANGES_REQUESTED** [review]

File: src/instruments/src/EFB/Store/performance-reducer.ts [hunk 2]
```diff
@@ -86,4 +95,4 @@ const Reducer = (state, action) => {
 // Curried
 export const PerformanceReducer = produce(Reducer);
 
-export const PerformanceContext:Context<TPerformanceContext> = createContext<TPerformanceContext>({ performanceState: performanceInitialState, performanceDispatch: () => {} });
+// export const PerformanceContext:Context<PerformanceContextType> = createContext<PerformanceContextType>({ performanceState: performanceInitialState, performanceDispatch: () => {} });
```
Comment (Benjozork) [inline src/instruments/src/EFB/Store/performance-reducer.ts:98]:
```suggestion
```

**Review by Benjozork — APPROVED** [review]

**Comment by Benjozork** [comment]
@antonyisoardi Can you please fill in the testing instructions ?

**Comment by St54Kevin** [comment]
**Quality Assurance Tester**

Object of testing: #5274 
Tier of Testing: 1
Date: 12/07/2021

**Testing Process:**
1. Loaded in and tested EFB all subpages

**Negatives:** N/A
**Testing Results:** Passed

**Comment by hiaaryan** [comment]
Please fix conflicts 😄

**Comment by TheSledge-SLEJ** [comment]
Thanks for working on this. Will be a fab milestone when complete.

**Comment by antonyisoardi** [comment]
@St54Kevin ready for retest please

@Benjozork @hiaaryan @TheSledge-SLEJ - refactored code ready for your review. Once this is in, I'll pick up the other state management stuff that @hiaaryan has raised for some of the other views and probably a bit of a refactor on the navigraph stuff to bring it inline with these views.

Cheers

**Comment by hiaaryan** [comment]
If it does not fix #5307 you can remove it from the description :)

File: src/instruments/src/EFB/Dispatch/Pages/FuelPage.tsx [hunk 3]
```diff
@@ -376,7 +376,7 @@ export const FuelPage = () => {
                     <div className="w-4/12 ml-3">
                         <div className="bg-navy-lighter rounded-2xl p-6 text-white shadow-lg overflow-x-hidden">
                             <h2 className="text-2xl font-medium">Right Outer Tank</h2>
-                            <div className="mt-4">
+                            <div className="flex mt-4">
```
Comment (hiaaryan) [inline src/instruments/src/EFB/Dispatch/Pages/FuelPage.tsx:379]: this should not have flex class.

File: src/instruments/src/EFB/Dispatch/Pages/FuelPage.tsx [hunk 4]
```diff
@@ -297,7 +297,7 @@ export const FuelPage = () => {
                     <div className="w-4/12 mr-3">
                         <div className="bg-navy-lighter rounded-2xl p-6 text-white shadow-lg overflow-x-hidden">
                             <h2 className="text-2xl font-medium">Left Outer Tank</h2>
-                            <div className="mt-4">
+                            <div className="flex mt-4">
```
Comment (hiaaryan) [inline src/instruments/src/EFB/Dispatch/Pages/FuelPage.tsx:300]: this should not have flex class.

**Review by hiaaryan — CHANGES_REQUESTED** [review]

File: src/instruments/src/EFB/Dispatch/Pages/FuelPage.tsx (hunk 4 above)
Comment (hiaaryan) [reply src/instruments/src/EFB/Dispatch/Pages/FuelPage.tsx:300]: This has not been removed still :D

**Review by hiaaryan — APPROVED** [review]

File: package.json [hunk 5]
```diff
@@ -46,6 +46,7 @@
     "lint-staged": "10.4.0",
     "postcss": "^8.3.5",
     "prettier": "2.1.2",
+    "postcss": "^8.3.5",
```
Comment (davidwalschots) [inline package.json:49]: **issue**: duplicate
```suggestion
```

**Review by davidwalschots — CHANGES_REQUESTED** [review]

**Comment by davidwalschots** [comment]
Can we get a clean commit history please?

File: package.json (hunk 5 above)
Comment (hiaaryan) [reply package.json:49]: wow, sorry, missed it somehow 😅

**Review by Benjozork — CHANGES_REQUESTED** [review]
Ok I'm sorry for saying this now, but we cannot keep adding more Redux stuff. This should be done with react-router, properly.

Either we wait until another PR migrates the EFB to react-router or we make this PR switch it.

**Comment by antonyisoardi** [comment]
> Ok I'm sorry for saying this now, but we cannot keep adding more Redux stuff. This should be done with react-router, properly.
> 
> Either we wait until another PR migrates the EFB to react-router or we make this PR switch it.

React Router will not give you state persistence. So what is it you want to achieve by using router?

**Comment by Benjozork** [comment]
I would like the routing system to be refactored before we do this. I didn't mean doing this with react router.

**Comment by Benjozork** [comment]
We have too many problems in the EFB right now that require refactors, especially regarding Redux, so I'd like to start ASAP. It's gonna block us very soon.

**Comment by Benjozork** [comment]
Here are the main issues.

- There's unclear use of Redux and sometimes uses of it that I feel are unjustified.

- There are complex components that seem to either be overengineered or badly designed.

- The routing system is currently relying on react state. It's very messy.

I think the third one is where we should start.

**Comment by davidwalschots** [comment]
To add to this. The mixing of module systems and JS/TS isn't good either. I guess this all stems from the fact that the EFB, and all other instruments are in the same "project" instead of being cleanly separated.

I would have liked to use a provider I built as part of the failures package:
```typescript
import { FailuresOrchestratorProvider } from '@flybywiresim/failures';
```
but I can't do that, because the file in which I want to use it is `index.jsx` instead of `index.tsx`. Changing to `tsx` brings its own problems as there is still a `js` import:
```typescript
import { getSimVar, setSimVar } from '../util.js';
```

Now I will have to figure out a way to deal with this, and I will, because I want to complete implementing the bare bones of failures. It is however a needlessly complex and time consuming affair. 😀

**Comment by Benjozork** [comment]
We should definitely remove the JS files in `src/` I think. Unfortunately the PFD still uses them. We could duplicate those functions to a TS module for now?

**Comment by Benjozork** [comment]
Maybe an issue should be opened for this!

**Comment by Venorcis** [comment]
> If it does not fix #5307 you can remove it from the description :)

This pull indeed does not seem to fix the chartsview persisting when you navigate, or am I missing something?

**Comment by 2hwk** [comment]
I will be moving this as under the 0.8.0 milestone, as we are planning to release 0.7.0 fairly soon.

**Comment by oim1** [comment]
Quality Assurance Trainee Report

Object of testing: (#5274)
Tier of Testing: 1
Date: 22/08/2021

Testing Process: 
Loaded Cold and Dark at YMML
Turned on Flypad
Copied simbrief flightplan
Clicked on Dispatch > OFP
Clicked onto another page 
Went back into Dispatch Page
Checked to see if it remained on OFP. 
Repeated for perf page. 

Negatives: It didn't remain on any subpage when I clicked back on.
Testing Results: Not Passed

Media: 

https://user-images.githubusercontent.com/81839029/130339662-087128ef-ffbd-4196-ba20-0cdfc041d713.mp4

**Comment by oim1** [comment]
Quality Assurance Trainee Report

Object of testing: (#5274)
Tier of Testing: 1
Date: 23/08/2021

Testing Process:
Loaded Cold and Dark at YMML
Turned on Flypad
Copied simbrief flightplan
Clicked on Dispatch > OFP
Clicked onto another page
Went back into Dispatch Page
Checked to see if it remained on OFP.
Repeated for perf page.

Negatives: N/A
Testing Results: Passed

Conclusion: Both performance and dispatch page remained on their subpage after moving away from them. Not sure what happened in the previous test but a reinstall of msfs seems to have fixed it. LGTM.

**Comment by Benjozork** [comment]
Please fix conflicts.

**Comment by ErickSharp** [comment]
This is being done in Flypad OS 3

**Pull request closed without merging**